src/model.rs

```diff
@@ -1438,6 +1571,249 @@ mod tests {
         }
     }
 
+    mod enhance_pickup_dropoff {
```

Comment:
Not sure (as I don't get the need for a specific module here):
```suggestion
    mod enhance_pickup_dropoff_tests {
```

Reply:
The only goal of these modules is to regroup tests by category (all the tests in this module relate to the algorithm about pick-up and drop-off). We could suffix with `_tests` but:
- it's not the convention we've used so far in other similar modules
- this module is already wrapped in a module `tests` so it's namespaced